src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1090,61 +1087,117 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                     ? receiver.Type
                     : this.ContainingType;
                 method = new ErrorMethodSymbol(methodContainer, returnType, name);
-                args = BuildArgumentsForErrorRecovery(analyzedArguments);
             }
 
+            // until we've considered the extension methods, we don't rewrite the arguments.
+            args = extensionMethodsOfSameViabilityAreAvailable
+                ? analyzedArguments.Arguments.ToImmutable()
+                : BuildArgumentsForErrorRecovery(analyzedArguments, methods);
+
             var argNames = analyzedArguments.GetNames();
             var argRefKinds = analyzedArguments.RefKinds.ToImmutableOrNull();
             return BoundCall.ErrorCall(node, receiver, method, args, argNames, argRefKinds, isDelegate, invokedAsExtensionMethod: invokedAsExtensionMethod, originalMethods: methods, resultKind: resultKind);
         }
 
-        private static ImmutableArray<BoundExpression> BuildArgumentsForErrorRecovery(AnalyzedArguments analyzedArguments, ImmutableArray<ParameterSymbol> parameters)
+        private bool IsUnboundGeneric(MethodSymbol method)
```

Comment:
I'll be happy to help you navigate the source offline.

---
In reply to: [67733199](https://github.com/dotnet/roslyn/pull/11981#discussion_r67733199) [](ancestors = 67733199,67732085)